docs/develop/sdks/nodejs/observability.mdx

```diff
@@ -0,0 +1,138 @@
+
+  // Set up metrics via prometheus
+  const metricsExporter = new PrometheusExporter({});
+
+  const myServiceMeterProvider = new MeterProvider({
+    resource: resource,
+  });
+
+  myServiceMeterProvider.addMetricReader(metricsExporter);
+
+  metrics.setGlobalMeterProvider(myServiceMeterProvider);
+```
+
+Then, create a middleware that captures the metric:
+```typescript
+import {Metadata, StatusObject} from '@grpc/grpc-js';
+import {Middleware, MiddlewareRequestHandler} from '@gomomento/sdk';
+import {metrics} from '@opentelemetry/api';
+import {Counter} from '@opentelemetry/api/build/src/metrics/Metric';
+import {Message} from 'google-protobuf';
+
+class ExampleMetricMiddlewareRequestHandler implements MiddlewareRequestHandler {
+  private requestCounter: Counter;
+  constructor(requestCounter: Counter) {
+    this.requestCounter = requestCounter;
+  }
+  onRequestMetadata(metadata: Metadata): Promise<Metadata> {
+    return Promise.resolve(metadata);
+  }
+
+  onRequestBody(request: Message): Promise<Message> {
+    const requestType = request.constructor.name;
+    this.requestCounter.add(1, {'request.type': requestType});
+    return Promise.resolve(request);
+  }
+
+  onResponseMetadata(metadata: Metadata): Promise<Metadata> {
+    return Promise.resolve(metadata);
+  }
+
+  onResponseBody(response: Message | null): Promise<Message | null> {
+    return Promise.resolve(response);
+  }
+
+  onResponseStatus(status: StatusObject): Promise<StatusObject> {
+    return Promise.resolve(status);
+  }
+}
+
+export class ExampleMetricMiddleware implements Middleware {
+  private readonly requestCounter: Counter;
+  constructor() {
+    const meter = metrics.getMeter('metric-middleware-meter');
+
+    this.requestCounter = meter.createCounter('momento_requests_counter', {
+      description: 'Momento GRPC requests',
+    });
+  }
+
+  onNewRequest(): MiddlewareRequestHandler {
+    return new ExampleMetricMiddlewareRequestHandler(this.requestCounter);
+  }
+}
+```
+
+When you create the Momento CacheClient, add the middleware and the metric will be incremented with each request:
+```typescript
+const momento = new CacheClient({
+    configuration: Configurations.Laptop.v1().withMiddlewares([new ExampleMetricMiddleware()]),
+    credentialProvider: CredentialProvider.fromEnvironmentVariable({
+      environmentVariableName: 'MOMENTO_AUTH_TOKEN',
+    }),
+    defaultTtlSeconds: 60,
+  });
+```
+
+## Traces
+Traces provide a detailed timeline of processes within an application, showing the relationship between different components and services involved in a specific request or operation. They allow developers to analyze the sequence and duration of these operations, facilitating a better understanding of how data flows through the system.
+
+The Momento Node.js SDK uses gRPC internally to communicate with the Momento servers. Here is an example that uses OpenTelemetry to automatically generate traces for these calls and exports them to Zipkin:
```

Comment:
I would consider adding a sentence here calling out that the tracing is auto-instrumented, something like:

```
The OpenTelemetry provides a capability for auto-instrumenting all gRPC calls with traces; you don't need to add any middleware code to produce these traces, like you do for the metrics.
```